### PR TITLE
enhancement #696: moved the endpoints for changing captain

### LINKED ISSUE #696: Move the endpoints for changing captain and change the interface
As we can see, these endpoints are in the user's service, which is by mistake. Move the endpoints shown below to the **group controller and group service**. Instead of id of the previous captain and the new one, you should receive groupId and new capatainId.

image
image

They differ only in permission and purpose. The first endpoint is created for captain himself to transfer the role, and the other for administration (for force changes, this means less checks).

### PR DESCRIPTION
**Pull request by f-ogoog**
- moved the endpoints for changing captain
- changed params
- moved and changed tests

closes #696

### CODE REVIEW

File: src/v2/api/controllers/GroupController.ts [hunk 1]
```diff
@@ -423,4 +424,57 @@ export class GroupController {
     return { students };
   }
 
+  @Access(PERMISSION.GROUPS_$GROUPID_TRANSFER)
+  @ApiBearerAuth()
+  @Post('/:groupId/transferCaptain')
+  @ApiOkResponse()
+  @ApiBadRequestResponse({
+    description: `\n
+     InvalidEntityIdException:
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/GroupController.ts:433]: Remove unnecessarily space. "I" symbol should be under description

File: src/v2/api/controllers/GroupController.ts [hunk 2]
```diff
@@ -423,4 +424,57 @@ export class GroupController {
...
+    description: `\n
+    UnauthorizedException:
+      Unauthorized`,
+  })
+  @ApiForbiddenResponse({
+    description: `\n
+    NoPermissionException:
+      You do not have permission to perform this action`,
+  })
+  async transferRole (
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/GroupController.ts:447]: Rename `transferRole` to `transferCaptain`

File: src/v2/api/services/GroupService.ts [hunk 3]
```diff
@@ -284,4 +285,63 @@ export class GroupService {
...
+          },
+          data: {
+            studentId,
+          },
+        },
+      },
+    });
+  }
+
+  private async checkIsStudentBelongToGroup (groupId: string, studentId: string) {
```
Comment (SharpDevOps10) [inline src/v2/api/services/GroupService.ts:342]: This method has long name, thus, it is better to rename `checkIsStudentBelongToGroup` to `isStudentInGroup`

File: src/v2/api/services/GroupService.ts [hunk 4]
```diff
@@ -284,4 +285,63 @@ export class GroupService {
     }
   }
 
+  async switchCaptain (groupId: string, studentId: string) {
+    const isStudentBelongToGroup = await this.checkIsStudentBelongToGroup(groupId, studentId);
```
Comment (SharpDevOps10) [inline src/v2/api/services/GroupService.ts:289]: Hence, rename this variable to `studentIsInGroup`

**Review by SharpDevOps10 — CHANGES_REQUESTED** [review]

**Review by SharpDevOps10 — APPROVED** [review]

**Review by ArtemLyah — APPROVED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**